rust/headless-client/src/linux.rs

```diff
@@ -10,31 +10,62 @@ use connlib_shared::{
 use firezone_cli_utils::setup_global_subscriber;
 use futures::{SinkExt, StreamExt};
 use secrecy::SecretString;
-use std::{
-    future,
-    net::IpAddr,
-    path::{Path, PathBuf},
-    str::FromStr,
-    task::Poll,
-};
+use std::{future, net::IpAddr, path::PathBuf, str::FromStr, task::Poll};
 use tokio::{
     net::{UnixListener, UnixStream},
     signal::unix::SignalKind,
 };
 use tokio_util::codec::LengthDelimitedCodec;
 
+const SOCK_PATH: &str = "/run/firezone-client.sock";
+
 pub async fn run() -> Result<()> {
     let cli = Cli::parse();
     let (layer, _handle) = cli.log_dir.as_deref().map(file_logger::layer).unzip();
     setup_global_subscriber(layer);
 
     match cli.command() {
+        Cmd::Auto => {
+            if let Some(token) = token(&cli).await? {
+                run_standalone(cli, &token).await
+            } else {
+                run_daemon(cli).await
+            }
+        }
         Cmd::Daemon => run_daemon(cli).await,
-        Cmd::Standalone => run_standalone(cli).await,
+        Cmd::DebugIpcClient => run_debug_ipc_client(cli).await,
+        Cmd::Standalone => {
+            let token = token(&cli)
+                .await?
+                .context("Need a token to run as standalone Client")?;
+            run_standalone(cli, &token).await
+        }
+    }
+}
+
+async fn token(cli: &Cli) -> Result<Option<SecretString>> {
+    if let Some(token) = &cli.token {
+        // Token was provided in CLI args or env var
```

Comment:
Perhaps log a `debug!` log that we are taking the token from the CLI / env var. It might be useful in debugging this to know why the token in `/etc` is ignored.